### PR TITLE
bpo-42470: Do not warn on sequences which are also sets in random.samples

### PR DESCRIPTION
**Pull request by masklinn**
https://bugs.python.org/issue42470

### CODE REVIEW

File: Lib/test/test_random.py [hunk 1]
```diff
@@ -163,6 +163,24 @@ def test_sample_on_sets(self):
             population = {10, 20, 30, 40, 50, 60, 70}
             self.gen.sample(population, k=5)
 
+    def test_sample_on_seqsets(self):
+        class SeqSet(abc.Sequence, abc.Set):
+            def __init__(self, items=()):
+                self._items = list(dict.fromkeys(items, None))
```
Comment (serhiy-storchaka) [inline Lib/test/test_random.py]: Why not just `self._items = items` and pass a list with unique items?

File: Lib/test/test_random.py [hunk 2]
```diff
@@ -163,6 +163,24 @@ def test_sample_on_sets(self):
...
+
+            def __len__(self):
+                return len(self._items)
+
+            def __getitem__(self, index):
+                return self._items[index]
+
+        population = SeqSet([1, 2, 3, 4, 1])
+        with warnings.catch_warnings(record=True) as ws:
+            warnings.simplefilter("always", DeprecationWarning)
```
Comment (serhiy-storchaka) [inline Lib/test/test_random.py]: Just make it "error". And no recording is needed.

File: Lib/test/test_random.py (hunk 1 above)
Comment (masklinn) [reply Lib/test/test_random.py]: Habit (to ensure the correct invariant), you're right that it's completely unnecessary for something which only exists for the purpose of a test.

File: Lib/random.py [hunk 3]
```diff
@@ -424,13 +424,14 @@ def sample(self, population, k, *, counts=None):
...
 
-        if isinstance(population, _Set):
-            _warn('Sampling from a set deprecated\n'
-                  'since Python 3.9 and will be removed in a subsequent version.',
-                  DeprecationWarning, 2)
-            population = tuple(population)
         if not isinstance(population, _Sequence):
-            raise TypeError("Population must be a sequence.  For dicts or sets, use sorted(d).")
+            if isinstance(population, _Set):
+                _warn('Sampling from a set deprecated\n'
```
Comment (serhiy-storchaka) [inline Lib/random.py:429]: Why add `\n` in the middle of a sentence?

File: Misc/NEWS.d/next/Library/2020-12-06-12-00-00.bpo-42470.iqtC4L.rst [hunk 4]
```diff
@@ -0,0 +1 @@
+random.sample no longer warns on a sequence which is also a set
```
Comment (serhiy-storchaka) [inline Misc/NEWS.d/next/Library/2020-12-06-12-00-00.bpo-42470.iqtC4L.rst]:
```suggestion
:func:`random.sample` no longer warns on a sequence which is also a set.
```

File: Lib/random.py (hunk 3 above)
Comment (masklinn) [reply Lib/random.py:429]: You’ll have to ask @rhettinger, that’s just a move / reindent of the existing warning.

**Review by serhiy-storchaka — APPROVED** [review]

**Review by rhettinger — APPROVED** [review]

**Pull request merged**